wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift

```diff
@@ -1014,6 +1028,31 @@ extension ZMUserSession: ZMSyncStateDelegate {
         }
     }
 
+    private func fetchBackendMLSPublicKeys() async {
+        do {
+            var getBackendMLSPublicKeysAction = FetchBackendMLSPublicKeysAction()
+            let backendPublicKeys = try await getBackendMLSPublicKeysAction.perform(in: notificationContext)
+            let hasValidKeys = backendPublicKeys.removal.hasValidKeys()
+            BackendInfo.isMLSEnabled = hasValidKeys
+        } catch {
+            WireLogger.mls.error("Backend doesn't have MLS public keys: \(String(reflecting: error))")
+        }
+    }
+
+    private func createMLSClient(client: UserClient) async {
+        let mlsClientID = await syncContext.perform {
+            return MLSClientID(userClient: client)
+        }
+        guard let mlsClientID else {
+            fatalError("Needs to register MLS client but can't retrieve qualified client ID")
+        }
+        do {
+            try await self.coreCryptoProvider.initialiseMLSWithBasicCredentials(mlsClientID: mlsClientID)
+        } catch {
+            WireLogger.mls.error("Failed to initialise mls client: \(error)")
```

Comment:
This would not be a nice state to be in.... we should have an mls client but we couldn't create one. What do you think would be the appropriate response here? thoughts @netbe ?

Reply:
I talked with Jacob, and his opinion:
`I would expect you to log that error and then try it again after the next quick sync.`

Reply:
great